Additional documentation (#7)
Jedimaster4559/PartwritingErrorChecker: docs/EffectiveUse.md

## What this software does well
This tool can provide instant feedback on a number of common part-writing errors. This can be particularly helpful when practicing part-writing as it can be hard for students to perform a self-assessment (or peer assessment) of their work.

> ## What it doesn't do well

Author: ```suggestion
## What it does not do well
```

> There are a couple of partwriting standards that this tool cannot assess. Most of these deal are related to music concepts that are much more complex from a computational perspective. For example, this software is not capable of detecting errors resolving the leading tone, since that would require the tool to have a concept of the current key area.

Author: ```suggestion
There are a couple of part-writing standards that this tool cannot assess. Most of these deal are related to music concepts that are much more complex from a computational perspective. For example, this software is not capable of detecting errors resolving the leading tone, since that would require the tool to have a concept of the current key area.
```

> This software is also incapable of any assessment of the musicality of the partwriting. There are often several valid solutions to any given partwriting assignment. This tool cannot tell any difference between those solutions, and which ones may have better or worse choices than others. It is entirely possible to complete partwriting that follows all the "rules" but doesn't sound good.

Author: ```suggestion
This software is also incapable of any assessment of the musicality of the part-writing. There are often several valid solutions to any given part-writing assignment. This tool cannot tell any difference between those solutions, and which ones may have better or worse choices than others. It is entirely possible to complete part-writing that follows all the "rules" but doesn't sound good.
```

At some point, fixes to these problems may be implemented in this software. See ![Future Work](#FutureWork) for more information and some proposed solutions.

## My biggest fear for this software
Many of the musicians I have met view music theory as just a course that needs to be completed to get their degree. I disagree with this perspective. If the course is taught well and students are willing to learn, music theory courses can have a great impact on developing a student's musicianship.

> This tool was built primarily to enable students to practice part writing independently. Ultimately, this tool aims to help students succeed in their coursework by allowing them to quickly analyze their own work. While this tool does a great job at taking a computational approach to partwriting and music theory, it cannot assess the musicality of a student's decisions, and is inflexible in situations where the typical part-writing rules can be broken. This is where the traditional grading and assessment of music theory takes a critical role. An experienced music theorist may choose to break the standard partwriting rules. In these situations, a theory instructor can provide much feedback regarding musical decision making that this software cannot.

Author: ```suggestion
This tool was built primarily to enable students to practice part writing independently. Ultimately, this tool aims to help students succeed in their coursework by allowing them to quickly analyze their own work. While this tool does a great job at taking a computational approach to part-writing and music theory, it cannot assess the musicality of a student's decisions, and is inflexible in situations where the typical part-writing rules can be broken. This is where the traditional grading and assessment of music theory takes a critical role. An experienced music theorist may choose to break the standard part-writing rules. In these situations, a theory instructor can provide much feedback regarding musical decision making that this software cannot.
```

> My biggest fear is that use of this software will hinder this transfer of musical knowledge and stifle the growth of students' musicianship. While this software could be an excellent tool for assiting instructor's grading processes, it should not be their only source of feedback to their students. Additionally, students should not rely on this tool as the sole assessment of their work. Ultimately this tool is just a tool. Critical thinking and true musicianship will always be more important than any of the capabilities of this software.

Author: ```suggestion
My biggest fear is that use of this software will hinder this transfer of musical knowledge and stifle the growth of students' musicianship. While this software could be an excellent tool for assisting instructor's grading processes, it should not be their only source of feedback to their students. Additionally, students should not rely on this tool as the sole assessment of their work. Ultimately this tool is just a tool. Critical thinking and true musicianship will always be more important than any of the capabilities of this software.
```
